src/backend/postgres/types.rs

```diff
@@ -35,7 +35,13 @@ impl TypeBuilder for PostgresQueryBuilder {
             write!(sql, "IF EXISTS ").unwrap();
         }
 
-        for name in drop.names.iter() {
+        let mut names_iter = drop.names.iter();
```

Comment:
I suggest another option:
```rust
drop.names.fold(true, |first, name| {
...
});
```
@PreetamSing what do you sink about it?)

Reply:
I cannot think of any way for using `fold` method without leaving an unnecessary comma after the last name, which in Postgres will throw an error. Like following:
```sql
DROP TYPE IF EXISTS "kyc_status", "font_family",  CASCADE;
```
Though, enlighten me if I'm wrong.

Reply:
I mean:
```rust
drop.names.fold(true, |first, name| {
    if !first {
       write!(sql ", ").unwrap();
    }
    self.prepare_type_ref(name, sql);
    false
});
```

Reply:
That works. Though, I think we are evaluating an **if** condition for every iteration, which is unnecessary and is saved from in my initial code.

Reply:
Hey @PreetamSing, I'm going to suggest something similar to what @ikrivosheev said.

In terms of performance, both implementation is roughly the same, with BigO `O(n)`. Benchmark: https://github.com/SeaQL/sea-query/commit/2bbfa34818091ed337c327929409da23b33ad380

```
     Running benches/basic.rs (target/release/deps/basic-2f2ede7235026e9f)

mut_iter                time:   [26.429 ms 26.527 ms 26.628 ms]

fold                    time:   [26.130 ms 26.203 ms 26.286 ms]

Found 8 outliers among 100 measurements (8.00%)
  5 (5.00%) high mild
  3 (3.00%) high severe
```

However, the `.fold(...)` implementation is optimized for readability and maintainability. So I'm more in favour of the refactoring.

Reply:
Umm... I like `iter().enumerate()` and `if i > 0` slightly more, just because it is more inline with the original code structure.
But either way is fine.